### PR TITLE
typescript-angular: fix oneOf and anyOf generates incorrect model for primitive types

### PR DESCRIPTION
**Pull request by SAnDAnGE**
fixes https://github.com/openAPITools/openapi-generator/issues/4130
fixes https://github.com/openAPITools/openapi-generator/issues/3149

### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [ ] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java [hunk 1]
```diff
@@ -562,22 +541,34 @@ public String toApiFilename(String name) {
         if (name.length() == 0) {
             return "default.service";
         }
+        if (importMapping.containsKey(name)) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java]: why is this necessary? I guess the filename should not contain paths

**Comment by macjohnny** [comment]
thanks for your pr
please follow the steps described in https://github.com/OpenAPITools/openapi-generator/blob/master/.github/PULL_REQUEST_TEMPLATE.md

especially, you need to re-generate the samples and commit the changes:
```
mvn clean package
bin/typescript-angular-petstore-all.sh
```

**Comment by SAnDAnGE** [comment]
Please merge, thank you.

**Comment by wing328** [comment]
@SAnDAnGE  thanks for the PR, which has been included in the v4.2.1 release: https://twitter.com/oas_generator/status/1195339336922759168

**Comment by gritschf** [comment]
@SAnDAnGE @wing328 Are you sure this fixes #3149 ([typescript-angular] Fix importMapping). Because in the constructor of `AbstractTypeScriptClientCodegen` the `importMapping` will still be cleared and in `TypeScriptAngularClientCodegen` will always be empty
